WooCommerce/Classes/Authentication/Jetpack Setup/Native Jetpack Setup/LoginJetpackSetupViewModel.swift

```diff
@@ -32,22 +35,130 @@ final class LoginJetpackSetupViewModel: ObservableObject {
         return attributedString
     }()
 
-    init(siteURL: String, connectionOnly: Bool, stores: StoresManager = ServiceLocator.stores) {
+    init(siteURL: String, connectionOnly: Bool, stores: StoresManager = ServiceLocator.stores, onStoreNavigation: @escaping () -> Void = {}) {
         self.siteURL = siteURL
         self.connectionOnly = connectionOnly
         self.stores = stores
         let setupSteps = connectionOnly ? [.connection, .done] : JetpackInstallStep.allCases
         self.setupSteps = setupSteps
-        self.currentSetupStep = setupSteps[0]
         self.title = connectionOnly ? Localization.connectingJetpack : Localization.installingJetpack
+        self.storeNavigationHandler = onStoreNavigation
     }
 
     func isSetupStepInProgress(_ step: JetpackInstallStep) -> Bool {
-        step == currentSetupStep && step != .done
+        guard let currentStep = currentSetupStep else {
+            return false
+        }
+        return step == currentStep && step != .done
     }
 
     func isSetupStepPending(_ step: JetpackInstallStep) -> Bool {
-        step > currentSetupStep
+        guard let currentStep = currentSetupStep else {
+            return false
+        }
+        return step > currentStep
+    }
+
+    func startSetup() {
+        retrieveJetpackPluginDetails()
+    }
+
+    func authorizeJetpackConnection() {
```

Comment:
Nit: Would it make sense to name it as `func didAuthorizeJetpackConnection()`? 

If I understand the flow correctly, this method is called after the user authorizes the Jetpack connection in a web view. The name `authorizeJetpackConnection` makes me assume that this method handles the authorization. 

WDYT?

Reply:
I was not sure about this name either - updated in 159aa165039198f2c2ab237edc209267d344ede3.